contrib/pkg/adm/managedns/enable.go

```diff
@@ -191,15 +206,59 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
 	return nil
 }
 
-func podHasEnvVar(p corev1.Pod, envVar string) bool {
-	for _, ev := range p.Spec.Containers[0].Env {
-		if ev.Name == envVar {
-			return true
+func getUpdatedManagedDNSConfig(dynClient client.Client, previousGeneration int64) (string, error) {
+	deployment := &appsv1.Deployment{}
+	key := types.NamespacedName{Name: hiveAdmissionDeployment, Namespace: hiveNamespace}
+
+	log.Info("waiting for new hiveadmission deployment")
+	for i := 0; i < 24; i++ {
+		err := dynClient.Get(context.TODO(), key, deployment)
```

Comment:
It would be nice if we got in the habit of using a watch for things like this instead of looping. See https://github.com/kubernetes/client-go/blob/1a26190bd76a9017e289958b9fba936430aa3704/tools/watch/until.go#L129.